### ISSUE TITLE
Recaptcha - v2 support

### ISSUE THREAD

**Issue by jesusbagpuss** [eprints/eprints#354]
See: #353 

Not sure if an update to https://github.com/eprints/eprints/blob/3.3/perl_lib/EPrints/MetaField/Recaptcha.pm or a new RecaptchaV2 metafield is best.
If the same keys are usable for both, the first option is probably better?

**Comment by phluid61** [comment]
I was experimenting a bit today.

The same key works in both versions, although I seem to recall the Google
page suggesting (recommending?) getting a new key. I'll follow up again
tomorrow.

For my testing I added a switch in the config, and branched in the renderer
and action. Maybe not the best way forward, especially if/when v3 comes out.

**Comment by phluid61** [comment]
Here's my code comparison; I won't open a PR until we do a bit more thinking about it in this issue.

**Comment by phluid61** [comment]
The [FAQ](https://developers.google.com/recaptcha/docs/faq?hl=en) suggests we register new keys when upgrading from v1.0 to 2.0, but I was able to switch between versions just fine using the same keys.

**Comment by jesusbagpuss** [comment]
Hi,
I've been reminded by Google that I need to stop using v1 of recaptcha.
I've tested your changes, mostly seems to work fine - good work!
One issue - on older versions of EPrints: `Can't locate object method "remote_ip" via package "EPrints::Repository" `

I wonder whether in the PR, we should remove all trace of v1 - as it will be mothballed March 31 2018?

**Comment by jesusbagpuss** [comment]
And this might be needed too:
```
if( EPrints::Utils::is_set( $ENV{HTTP_proxy} ) )
{
        $ua->proxy( 'http', $ENV{HTTP_proxy} )
}
```

**Comment by drn05r** [comment]
This is fixed with the pull request for #468, which will be merged in for 3.3.17.
